src_c/math.c

```diff
@@ -547,40 +549,47 @@ static PyMemberDef vector_members[] = {
 };
 
 static PyObject *
-pgVector_NEW(Py_ssize_t dim)
+_vector_subtype_new(PyTypeObject *type, Py_ssize_t dim)
 {
     pgVector *vec;
+    Py_ssize_t i;
+
     switch (dim) {
         case 2:
-            vec = PyObject_New(pgVector, &pgVector2_Type);
+            vec = (pgVector *)pgVector2_Type.tp_new(type, NULL, NULL);
             break;
         case 3:
-            vec = PyObject_New(pgVector, &pgVector3_Type);
+            vec = (pgVector *)pgVector3_Type.tp_new(type, NULL, NULL);
             break;
-            /*
-                case 4:
-                    vec = PyObject_New(pgVector, &pgVector4_Type);
-                    break;
-            */
         default:
             PyErr_SetString(PyExc_SystemError,
                             "Wrong internal call to pgVector_NEW.\n");
             return NULL;
     }
 
-    if (vec != NULL) {
+    if (vec) {
         vec->dim = dim;
-        vec->epsilon = VECTOR_EPSILON;
         vec->coords = PyMem_New(double, dim);
-        if (vec->coords == NULL) {
-            Py_DECREF(vec);
-            return PyErr_NoMemory();
-        }
```

Comment:
why was this check removed? It is very much needed, so it needs to be added back